site2/docs/functions-develop-api.md

```diff
@@ -102,6 +102,39 @@ public class ExclamationFunction implements Function<String, String> {
 
 For more details, see [code example](https://github.com/apache/pulsar/blob/master/pulsar-functions/java-examples/src/main/java/org/apache/pulsar/functions/api/examples/ExclamationFunction.java).
 
+The return type of the `Function` can be wrapped in a Pulsar `Record` generic which gives more control on the output message (topic, schema, properties, ...)
+You can use the `Context::newOutputRecordBuilder` method to build this `Record` output.
+
+```java
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.pulsar.functions.api.Context;
+import org.apache.pulsar.functions.api.Function;
+import org.apache.pulsar.functions.api.Record;
+
+public class RecordFunction implements Function<String, Record<String>> {
+
+    @Override
+    public Record<String> process(String input, Context context) throws Exception {
+        String publishTopic = (String) context.getUserConfigValueOrDefault("publish-topic", "publishtopic");
```

Comment:
This example is not good, with this example the function won't write to this topic (that is a  custom parameter)  but to the topic configured as output topic for the function.

We should rework the example

Reply:
The function *will* write to publishTopic. That's the interest of returning a Record 🙂

Reply:
In my opinion changing the destination (routing) is a secondary case, and it is not the core value of this feature.


I suggest to add a second paragraph about "routing", 
we should focus the first paragraph about the ability of setting a Schema, that is the main feature that was not available

Reply:
I removed the destinationTopic part.